core/src/avm2/globals/flash/net/URLStream.as

```diff
@@ -74,10 +80,12 @@ package flash.net {
 
         public function load(request:URLRequest):void {
             this._loader.load(request);
+            this._connected = true;
```

Comment:
I'm surprised it's this easy. Does it work this way even if the request fails (for example, the URLRequest url 404s)?

Reply:
Yes. `connected` seems to stay true no matter if the request succeeds or fails.